sql/updates/world/4.3.4/2021_01_21_01_world.sql

```diff
@@ -0,0 +1,1413 @@
+(43010,145,11793,0);
+
+-- add `creature_trainer` records for low level paladin trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (925,926,15280,37737);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES 
+(925,168,4663,0),
+(926,168,4678,0),
+(15280,168,0,3),
+(37737,168,11767,0);
+
+-- add `creature_trainer` records for low level hunter trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (895,3061,34673,35874,38247,39214,43278);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES 
+(895,15,4675,0),
+(3061,15,4647,0),
+(34673,15,0,3),
+(35874,15,0,3),
+(38247,15,10988,0),
+(39214,15,0,3),
+(43278,15,14200,0);
+
+-- add `creature_trainer` records for low level rogue trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (916,34693,35871,38244,42366);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES 
+(916,17,4676,0),
+(34693,17,0,3),
+(35871,17,0,3),
+(38244,17,0,3),
+(42366,17,0,3);
+
+-- add `creature_trainer` records for low level priest trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (837,2123,3595,15284,35870,38245,42323);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES 
+(837,127,4679,0),
+(2123,127,3645,0),
+(3595,127,4692,0),
+(15284,127,0,3),
+(35870,127,10700,0),
+(38245,127,10986,0),
+(42323,127,0,3);
+
+-- add `creature_trainer` records for low level shaman trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (3062,3157,17089,37115,38242);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES 
+(3062,134,4103,0),
+(3157,134,4652,0),
+(17089,134,7522,0),
+(37115,134,0,3),
+(38242,134,0,3);
+
+-- add `creature_trainer` records for low level mage trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (198,15279,16500,37121,38246,42331,43006);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES 
+(198,44,4660,0),
+(15279,44,0,3),
+(16500,44,7264,0),
+(37121,44,0,3),
+(38246,44,10987,0),
+(42331,44,0,3),
+(43006,44,0,3);
+
+-- add `creature_trainer` records for low level warlock trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (459,460,2126,3156,15283,34696,35869,42618,43455);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES 
+(459,32,1503,0),
+(460,32,4681,0),
+(2126,32,4655,0),
+(3156,32,4643,0),
+(15283,32,0,3),
+(34696,32,10681,0),
+(35869,32,0,3),
+(42618,32,0,3),
+(43455,32,0,3);
+
+-- add `creature_trainer` records for low level druid trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (35873);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES 
+(35873,50,10696,0);
+
+-- Correct existing creature_trainer records with correct trainers
+UPDATE `creature_trainer` SET `MenuId` = 13921 WHERE `CreatureId` = 375;
```

Comment:
this is by the way, not correct. I've checked retail videos and the human priest trainer as well as most other low level priest trainers, offer high level spells in the starting zones

Reply:
I believe I changed that Menu Id because the creature_template was already set to that in gossip_menu_id.  I have no way of knowing which one is right, so I just preferred the creature_template in most cases.

Do you have links to those videos?  I couldn't find any direct evidence regarding low level trainers v high level trainers.  I do agree that trainers in start ZONES (like Elwynn Forest or Durotar) offer the full spell list, but not all of them.  

I saw indirect evidence that trainers in start AREAS (like Northshire Abbey and Echo Isles) tend to offer spells for the first 5 levels only.
1. Spell lists exist in your database for low level trainers, and they go up to level 5.
2. The immediate start areas level players to around level 5 before sending them off into the zone at large.
3. Classic and Wrath both had that kind of progression so I assumed it carried over to Cata.

Reply:
I'm talking about Northshire. I have been checking the priest trainer in Northshire myself a while ago because you might have noticed that this trainer was working already so I was on it before.

Reply:
Weird.  Okay.  It's easy enough to fix.  Should I change all of the low level trainers to give them the full spell list or just the priest?

Reply:
Ah ha!  Found a video.  [https://www.youtube.com/watch?v=krdXwD1TAvg](https://www.youtube.com/watch?v=krdXwD1TAvg)

It shows the warrior trainer with only the first two spells.  This was taken during the Cata Beta, though.

Reply:
It's only valid for priest trainers surprisingly. Other trainers are sorted by that.

Reply:
Not every priest trainer, though: [https://www.youtube.com/watch?v=pRbmpmIFS5s](https://www.youtube.com/watch?v=pRbmpmIFS5s)  Check 19m37s.  

Gnome priests were new to Cata.  Maybe they missed the priest trainers that existed in Wrath?  So human, dwarf, nelf, undead.  Probably not the troll, that area was changed significantly for Cata.  Belf and Draenei zones were untouched in Cata, so however Wrath is set up is probably right.